types/logger.d.ts

```diff
@@ -11,14 +13,24 @@ export interface FastifyLogFn {
 
 export type LogLevels = 'info' | 'error' | 'debug' | 'fatal' | 'warn' | 'trace'
 
+export interface FastifyLoggerInstance {
+  info: FastifyLogFn;
+  warn: FastifyLogFn;
+  error: FastifyLogFn;
+  fatal: FastifyLogFn;
+  trace: FastifyLogFn;
+  debug: FastifyLogFn;
+  child: FastifyLoggerInstance;
+}
+
 /**
  * Fastify Custom Logger options.
  */
```

Comment:
At one point there was a comment here that explained what do for Pino types. Maybe we should add the back in if this is misleading.

Reply:
I added a new comment :)